### PR TITLE
Remove/unexport/inline various utility functions

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

This is a non-functional change to reduce the "utility" package, by removing, inlining, unexporting, or replacing various functions.

 - `Contains` is replaced by using maps
 - `ParseForwarded`/`ForwardedHeader` are unexported
 - `ErrFetch` is unexported, and the one external usage (a test) now inlines the expected error message
 - `Prune` is inlined in the one place it was used
 - `InsertInMap` was unused, and has been removed
 - `reverse` (internal function) has been replaced by simply iterating through the key parts in reverse order

## Checklist

- [x] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
~- [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~

I have considered changes for:
~- [ ] documentation~
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
- [x] automated tests (add tests for the code changes, all [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally)
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

`make test`

## Related issues

None.

### CODE REVIEW

File: idxmgmt/manager.go [hunk 1]
```diff
@@ -225,22 +224,26 @@ func (m *manager) loadEventTemplate(feature feature, ilmSupporter libilm.Support
...
 
-func (m *manager) loadPolicy(ilmFeature feature, ilmSupporter libilm.Supporter, policiesLoaded []string) ([]string, error) {
+func (m *manager) loadPolicy(ilmFeature feature, ilmSupporter libilm.Supporter, policiesLoaded map[string]bool) error {
+	if !ilmFeature.enabled {
+		return nil
+	}
 	policy := ilmSupporter.Policy().Name
-	if !ilmFeature.enabled || utility.Contains(policy, policiesLoaded) {
-		return policiesLoaded, nil
+	if policiesLoaded[policy] {
```
Comment (simitt) [inline idxmgmt/manager.go:232]: Why are you removing the `!ilmFeature.enabled` check here?

**Review by simitt — APPROVED** [review]
LGTM except for one comment.

File: idxmgmt/manager.go (hunk 1 above)
Comment (axw) [reply idxmgmt/manager.go:232]: It's not removed, I just separated the check above before calling `ilmFeature.Policy()`

Comment (simitt) [reply idxmgmt/manager.go:232]: two lines up 🤦‍♀️ - thanks

**Pull request merged**